### ISSUE TITLE
Add Firebird backend

### ISSUE THREAD

**Issue by iwbnwif** [mtangoo/wxDatabase#12]
I would like to propose implementing a Firebird backend based on IBPP. 

This would make IBPP a sort of dependency, but given it comes as open source .cpp and .h files it can live in /vendors.

If this is approved then it would be a thinish wrapper around IBPP rather than an update of the wxDatabaseLayer Firebird backend. In future IBPP might be replaced by the new Firebird 3 OO API, but we should wait for things to stabilize first.

**Comment by mtangoo** [comment]
are you willing to add and maintain that one? I don't use FB and have zero idea about it.
I will add it if you can make one and agree to maintain it

**Comment by iwbnwif** [comment]
I will add it first using the PR mechanism you described earlier. Once it is working and stable we can decide if it is viable to be added to the main code. I wanted to add this ticket (how do I set "Enhancement"?) so we can discuss it here :)

**Comment by mtangoo** [comment]
Yes, fork it and add it in your fork and make tests there. Once its working, you can do a PR
@manyleaves can you help here on unit tests thing. I'm zero at that :)
